api/nodes.go

```diff
@@ -435,6 +435,38 @@ func (n *Nodes) GcAlloc(allocID string, q *QueryOptions) error {
 	return err
 }
 
+// Purge removes a node from the system. Nodes can still re-join the cluster if
+// they are alive.
+func (n *Nodes) Purge(nodeID string, q *QueryOptions) (*NodePurgeResponse, *QueryMeta, error) {
+	var resp NodePurgeResponse
+	path := fmt.Sprintf("/v1/node/%s/purge", nodeID)
+	qm, err := n.client.putQuery(path, nil, &resp, q)
+	if err != nil {
+		return nil, nil, err
+	}
+	return &resp, qm, nil
+}
+
+// NodePurgeResponse is used to deserialize a Purge response.
+type NodePurgeResponse struct {
+	HeartbeatTTL    time.Duration
+	EvalIDs         []string
+	EvalCreateIndex uint64
+	NodeModifyIndex uint64
+	LeaderRPCAddr   string
+	NumNodes        int32
+	Servers         []*NodeServerInfo
+}
```

Comment:
Aw phooey I didn't understand the extent of the mismatch when we discussed this response struct earlier.

I think adding a new struct would be preferable with just the fields this endpoint uses. Since the same fields will still be populated it shouldn't cause any compatibility concerns.

The only other alternative I can think of to make this coherent is to fill in these fields which I just don't think is useful.

Reply:
Should I leave the agent HTTP API alone for now or change it as part of this work to only return the fields the RPC call populates? I don't want to make any backwards compatible changes, but am also concerned with creating a mismatch between what a user sees when performing node purge via CURL and via the SDK.

Reply:
We're only concerned with "binary" compatibility for API responses. What matters is the JSON output is stable. We can change the Go types that the JSON is generated from.

AFAICT we should be able to create a NodePurgeResponse struct that is the subset of NodeUpdateResponse fields that Purges actually use and no one will be the wiser that we switched types around internally.